### PR TITLE
Add error message for invalid argument value

### LINKED ISSUE #236: Misleading error message for invalid choice arguments
When an argument has a list of available choices and user provides an invalid entry, I suppose to see an error message related to that. Something that is aligned with `argparser` documentation:
```
game.py: error: argument move: invalid choice: 'fire' (choose from 'rock',
'paper', 'scissors')
```
However, Knack gives an error message like this `mycli abc str: 'fire' is not in the 'mycli abc str' command group. See 'mycli abc str --help'.`

<details>
  <summary>A minimum example (mycli.py), click to see the code</summary>
  
  ```python
import sys
from collections import OrderedDict

from knack import CLI, ArgumentsContext, CLICommandsLoader
from knack.commands import CommandGroup


def abc_str(move: str):
    return [f"Your move was: {move}"]


class MyCommandsLoader(CLICommandsLoader):
    def load_command_table(self, args):
        with CommandGroup(self, 'abc', '__main__#{}') as g:
            g.command('str', 'abc_str')
        return OrderedDict(self.command_table)

    def load_arguments(self, command):
        with ArgumentsContext(self, 'abc str') as ac:
            ac.argument("move", 
                options_list=("--move", "-m"),
                choices=["rock", "paper", "scissors"],
                type=str)
        super(MyCommandsLoader, self).load_arguments(command)


mycli = CLI(cli_name='mycli', commands_loader_cls=MyCommandsLoader)
exit_code = mycli.invoke(sys.argv[1:])
sys.exit(exit_code)
  ```
</details>

Run the above code like:
```shell
> python mycli.py abc str --move fire
mycli abc str: 'fire' is not in the 'mycli abc str' command group. See 'mycli abc str --help'.
```

Here is the link to the line that is related to the error https://github.com/microsoft/knack/blob/cd590312b9b4ea1402d7a0a76814f6ca4da7fab8/knack/parser.py#L268

Knack version: 0.7.2
Python: 3.8
OS: Windows 10 x64

### PR DESCRIPTION
**Pull request by jiasli**
Fix #236
Migrated from https://github.com/microsoft/knack/pull/237

## Symptom

Knack can only give error message for invalid command, but not invalid argument value. 
History: https://github.com/microsoft/knack/pull/237#issuecomment-801876101

## Change

Add error message for invalid argument value.

## Testing guide
```
> python examples\exapp2 demo arg1
exapp2 demo: 'arg1' is not in the 'exapp2 demo' command group. See 'exapp2 demo --help'.

The most similar choices to 'arg1':
        arg

> python D:\cli\knack\examples\exapp2 demo arg --move rok
exapp2 demo arg: 'rok' is not a valid value for '--move'.
Allowed values: rock, paper, scissors
```

ℹ As `Allowed values: rock, paper, scissors` appears in one line, it is not needed to add an empty line before it.

### CODE REVIEW

File: knack/parser.py [hunk 1]
```diff
@@ -266,19 +266,24 @@ def _check_value(self, action, value):
...
-                suggestion_msg += '\n'.join(['\t' + candidate for candidate in candidates])
+            if action.dest in ["_command", "_subcommand"]:
+                # Command
+                error_msg = "{prog}: '{value}' is not in the '{prog}' command group. See '{prog} --help'.".format(
+                    prog=self.prog, value=value)
+                logger.error(error_msg)
+                # Show suggestions
+                candidates = difflib.get_close_matches(value, action.choices, cutoff=0.7)
+                if candidates:
+                    suggestion_msg = "\nThe most similar choices to '{value}':\n".format(value=value)
```
Comment (jiasli) [inline knack/parser.py:277]: No need to play around with English grammar for [third person singular form](https://www.gymglish.com/en/gymglish/english-grammar/the-s-in-the-third-person-singular-form).

File: knack/parser.py [hunk 2]
```diff
@@ -266,19 +266,24 @@ def _check_value(self, action, value):
...
+                    suggestion_msg += '\n'.join(['\t' + candidate for candidate in candidates])
+                    print(suggestion_msg, file=sys.stderr)
+            else:
+                # Argument
+                error_msg = "{prog}: '{value}' is not a valid value for '{name}'.".format(
+                    prog=self.prog, value=value,
+                    name=argparse._get_action_name(action))  # pylint: disable=protected-access
+                logger.error(error_msg)
+                # Show all allowed values
+                suggestion_msg = "Allowed values: " + ', '.join(action.choices)
```
Comment (jiasli) [inline knack/parser.py:287]: The number of allowed argument values is usually very limited. Show all of them.

**Review by fralik — APPROVED** [review]
Looks good to me!

**Review by fralik — APPROVED** [review]

File: knack/parser.py (hunk 1 above)
Comment (fralik) [reply knack/parser.py:277]: 👍

File: knack/parser.py (hunk 2 above)
Comment (fralik) [reply knack/parser.py:287]: 👍

**Review by fralik — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]
LGTM

**Pull request merged**